src/Affjax.purs

```diff
@@ -215,6 +218,8 @@ request driver req =
           TimeoutError
         else if message == requestFailedMessageIdent then
           RequestFailedError
+        else if message == "Request path contains unescaped characters" then
+          BadUrlError req.url
```

Comment:
It seems to me like `RequestContentError` would work for this case?

Then we would not have to add `BadUrlError` and then we would not have to bump the major version.